packages/backend/src/modules/tvl/TvlCleaner.ts

```diff
@@ -0,0 +1,81 @@
+import { Logger } from '@l2beat/backend-tools'
+import { UnixTime } from '@l2beat/shared-pure'
+
+import { Config } from '../../config/Config'
+import { Clock } from '../../core/Clock'
+import { TaskQueue } from '../../core/queue/TaskQueue'
+import { TvlDatabase } from './types'
+
+interface Table {
+  deleteHourlyUntil: (timestamp: UnixTime) => Promise<number>
+  deleteSixHourlyUntil: (timestamp: UnixTime) => Promise<number>
+}
+
+export class TvlCleaner {
+  private readonly taskQueue: TaskQueue<void>
+
+  constructor(
+    private readonly clock: Clock,
+    private readonly logger: Logger,
+    private readonly tables: Table[],
+  ) {
+    this.logger = this.logger.for(this)
+    this.taskQueue = new TaskQueue(
+      () => this.clean(),
+      this.logger.for('taskQueue'),
+      {
+        metricsId: TvlCleaner.name,
+      },
+    )
+  }
+
+  start() {
+    this.taskQueue.addToFront()
+    this.logger.info('Started')
+    return this.clock.onNewHour(() => {
+      this.taskQueue.addToFront()
+    })
+  }
+
+  async clean() {
+    const hourlyDeletionBoundary =
+      this.clock._TVL_ONLY_getHourlyDeletionBoundary()
+    const sixHourlyDeletionBoundary =
+      this.clock._TVL_ONLY_getSixHourlyDeletionBoundary()
+
+    for (const table of this.tables) {
```

Comment:
Maybe we can use Promise.all here? Those are separate tables after all

Reply:
I am a bit afraid of over-using our DB

Reply:
This will run alongside our regular backend, I do not want it to take all the connections for some time.